hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -466,14 +466,14 @@ public Pair<Option<String>, JavaRDD<WriteStatus>> syncOnce() throws IOException
     try {
       // Refresh Timeline
       HoodieTableMetaClient metaClient = initializeMetaClientAndRefreshTimeline();
-      String instantTime = metaClient.createNewInstantTime();
 
-      Pair<InputBatch, Boolean> inputBatchAndUseRowWriter = readFromSource(instantTime, metaClient);
+      Pair<InputBatch, Boolean> inputBatchAndUseRowWriter = readFromSource(metaClient);
 
       if (inputBatchAndUseRowWriter != null) {
         InputBatch inputBatch = inputBatchAndUseRowWriter.getLeft();
         boolean useRowWriter = inputBatchAndUseRowWriter.getRight();
         initializeWriteClientAndRetryTableServices(inputBatch, metaClient);
+        String instantTime = startCommit(metaClient, !autoGenerateRecordKeys);
```

Comment:
It looks like the `instantTime` is only used by `writeToSinkAndDoMetaSync`.  So the `metaClient` can be passed to `writeToSinkAndDoMetaSync` so all commit and write logic is in one place?

Reply:
Makes sense, updating now